cpp/include/cudf/utilities/error.hpp

```diff
@@ -101,9 +119,16 @@ namespace detail {
 
 inline void throw_cuda_error(cudaError_t error, const char* file, unsigned int line)
 {
-  throw cudf::cuda_error(std::string{"CUDA error encountered at: " + std::string{file} + ":" +
-                                     std::to_string(line) + ": " + std::to_string(error) + " " +
-                                     cudaGetErrorName(error) + " " + cudaGetErrorString(error)});
+  cudaGetLastError();
```

Comment:
Let's add the doc I wrote about why this algorithm is doing what it's doing.

Reply:
Doc added